app/src/main/java/com/example/demoappkotlin/WebServices/RecyclerViewWithApi.kt

```diff
@@ -62,12 +65,18 @@ class RecyclerViewWithApi : AppCompatActivity() {
                     dataClass = gson.fromJson(response, ModelRecyclerWithApi::class.java)
                     progressBarForRecycler.visibility = View.GONE
                     Log.d("Pretty Printed JSON :", dataClass.toString())
-                    val adapter = RecyclerViewWithApiAdapter(dataClass.data)
+                    val adapter = RecyclerViewWithApiAdapter(dataClass.data,this@RecyclerViewWithApi)
                     recyclerViewWithApi.adapter = adapter
                 }
             } else {
                 Log.e("HTTPURLCONNECTION_ERROR", responseCode.toString())
             }
         }
     }
+
+    override fun onUserClicked(position: Int) {
+        val intent = Intent(this,SingleUserActivity::class.java)
+        intent.putExtra("Selected User",position)
```

Comment:
Use apply. make sure that you check comment in all the files. Previously added this comment in other file.